azurerm/data_source_arm_scheduled_time.go

```diff
@@ -0,0 +1,171 @@
+package azurerm
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+type Frequency string
+
+const (
+	Day     Frequency = "Day"
+	Hour    Frequency = "Hour"
+	Month   Frequency = "Month"
+	OneTime Frequency = "OneTime"
+	Week    Frequency = "Week"
+)
+
+func dataSourceArmScheduledTime() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmScheduledTimeRead,
+		Schema: map[string]*schema.Schema{
+			"frequency": {
+				Type:             schema.TypeString,
+				Required:         true,
+				DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(Day),
+					string(Hour),
+					string(Month),
+					string(OneTime),
+					string(Week),
+				}, true),
+			},
+			"second": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"minute": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"hour": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"day_of_week": {
+				Type:          schema.TypeInt,
+				Optional:      true,
+				ConflictsWith: []string{"day_of_month"},
+			},
+			"day_of_month": {
+				Type:          schema.TypeInt,
+				Optional:      true,
+				ConflictsWith: []string{"day_of_week"},
+			},
+			"minimum_delay_from_now_in_minutes": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"timezone": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"next_run_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceArmScheduledTimeRead(d *schema.ResourceData, meta interface{}) error {
+
+	var shiftTime int
+	var location *time.Location
+
+	if v, exists := d.GetOkExists("minimum_delay_from_now_in_minutes"); exists {
+		shiftTime = v.(int)
+	} else {
+		shiftTime = 0
+	}
```

Comment:
we shouldn't need the `else` here - since it's an Integer it'll be `0` by default (and exists will always be true)

Reply:
Done!